File: python/power2.c
#include <stdio.h>

void power2(int i)
{
   int products[10];
   int x = 1;
   int c = 0;
   while(i>0)
   {
      if(i%2)
      {
         products[c++]=x;
      }
      x=x*2; // Multiply by 2
      i=i>>1; // Do a shift. Is this a bug?
   }
   for(x=0; x< c; x++)
   {
      printf("%i\n",products[x]);
   }

}


void main(int argc, char* argv[])
{
    int value = atoi(argv[1]);
    power2(value);
}

File: python/power2.py
#!/usr/bin/python

import sys

def power2(i):
    products = []
    x = 1
    while (i > 0):
        if (i % 2):
            products.insert(0, x)
        x = x * 2  # Multiply by 2
        i = i >> 1 # Do a shift. Is this a bug?
    # Print the powers.
    for x in products:
        print x
    return products

if __name__ == '__main__':
    value = int(sys.argv[1])
    power2(value)
